### PR TITLE
[[ Bug 20259 ]] Fix crash when inserting large binary data to SQLite databases that aren't opened with the "binary" option

### PR DESCRIPTION
**Pull request by techstrategies**

### CODE REVIEW

File: revdb/src/sqlite_connection.cpp [hunk 1]
```diff
@@ -439,7 +439,7 @@ bool queryCallback(void *p_context, int p_placeholder, DBBuffer& p_output)
 		else
 		{
 			// According to documentation in sqlitedecode.cpp, this is the required size of output buffer
-			t_escaped_string = malloc(2 + (257 * t_parameter_value . length) / 254);
+			t_escaped_string = malloc(2 + (257 * (long)t_parameter_value . length) / 254);
```
Comment (runrevmark) [inline revdb/src/sqlite_connection.cpp]: You need to use int64_t here and not 'long' as the size of long depends on processor bitness and arch (32-bit non-Windows - long is 32-bit, 64-bit non-Windows - long is 64-bit; 32-bit Windows - long is 32-bit, 64-bit Windows - long is 32-bit).

**Comment by runrevmark** [comment]
@techstrategies : Good catch! Thanks for the PR... Just a small comment about the data type to use which will fix the bug on all arch/os combinations we support :)

File: revdb/src/sqlite_connection.cpp (hunk 1 above)
Comment (techstrategies) [reply revdb/src/sqlite_connection.cpp]: Ah yeah.... good point :-)

**Comment by runrevmark** [comment]
@techstrategies Could you add a test for this too? Might need some advice from @livecodeali :)

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 78da768

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 78da76880a56bf25c0404d64ea483f4a1ca19a59

**Comment by livecode-vulcan** [comment]
😞 test failure 78da76880a56bf25c0404d64ea483f4a1ca19a59

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/613)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/629)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/603)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/546)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/549)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/607)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/627)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/612)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/629)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/611)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/622)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/644)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/608)

**Comment by runrevmark** [comment]
@techstrategies Just to be irksome - Win32 doesn't have int64_t by default - I *think* you can `#include <stdint.h>` in dbsqlite_connection.cpp to solve the problem though!

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 044e30a

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 044e30ad1f2b59bba7c63c418b4e4c3df0bc1b9f

**Comment by livecode-vulcan** [comment]
😎 test success 044e30ad1f2b59bba7c63c418b4e4c3df0bc1b9f

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/631)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/647)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/621)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/564)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/567)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/625)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/645)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/630)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/647)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/629)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/640)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/662)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/626)

**Pull request merged**